assignment-client/src/avatars/AvatarMixerSlavePool.h

```diff
@@ -95,19 +95,20 @@ class AvatarMixerSlavePool {
     friend bool AvatarMixerSlaveThread::try_pop(SharedNodePointer& node);
 
     // synchronization state
-    Mutex _mutex;
-    ConditionVariable _slaveCondition;
+    Mutex _poolMutex;
     ConditionVariable _poolCondition;
+    Mutex _slaveMutex; // subservient to _poolMutex, do not lock _poolMutex while holding _slaveMutex!
+    ConditionVariable _slaveCondition;
     void (AvatarMixerSlave::*_function)(const SharedNodePointer& node);
     std::function<void(AvatarMixerSlave&)> _configure;
 
     // Set from Domain Settings:
     float _priorityReservedFraction { 0.4f };
     int _numThreads { 0 };
 
-    int _numStarted { 0 }; // guarded by _mutex
-    int _numFinished { 0 }; // guarded by _mutex
-    int _numStopped { 0 }; // guarded by _mutex
+    int _numStarted { 0 }; // guarded by _slaveMutex
+    int _numFinished { 0 }; // guarded by _poolMutex
+    int _numStopped { 0 }; // guarded by _poolMutex
```

Comment:
for a separate PR: I haven’t looked at the rest of these classes too much but could we DRY up this code?  and I’ve suggested this before but there are lots of alternatives to the whole “slave”/“master” metaphor that are less gross, I’d love if we did a cleanup pass on that

Reply:
Suggestions welcome I guess, we have a single "pool" object that sets up a number of properties before flagging a number of threads to wake up, do a unit of work, and fall asleep again.  The code between the two files is similar enough that we could refactor it out, albeit likely into template classes.

Once we get farther on the PR review (or actually pick which one we'd like to go with) I see no problem with renaming variables (and class names)